tck/src/main/java/org/eclipse/microprofile/concurrency/tck/cdi/ManagedExecutorSharingBean.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.concurrency.tck.cdi;
+
+import static org.testng.Assert.*;
+
+import java.util.Arrays;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.concurrent.ManagedExecutor;
+import org.eclipse.microprofile.concurrent.ManagedExecutorConfig;
+import org.eclipse.microprofile.concurrent.NamedInstance;
+
+@RequestScoped
+public class ManagedExecutorSharingBean {
+
+    @Inject
+    ManagedExecutor defaultME;
+
+    @Inject
+    ManagedExecutor defaultME2;
+
+    @Inject
+    @ManagedExecutorConfig
+    ManagedExecutor defaultAnno1;
+
+    @Inject
+    @ManagedExecutorConfig
+    ManagedExecutor defaultAnno2;
+
+    @Inject
+    @NamedInstance("A")
+    @ManagedExecutorConfig
+    ManagedExecutor namedExecA1;
+
+    @Inject
+    @NamedInstance("A")
+    ManagedExecutor namedExecA2;
+
+    @Inject
+    @NamedInstance("B")
+    @ManagedExecutorConfig
+    ManagedExecutor namedExecB1;
+
+    @Inject
+    @NamedInstance("B")
+    ManagedExecutor namedExecB2;
+
+    /**
+     * Verify the container creates a ManagedExecutor instance per unqualified ManagedExecutor injection point
+     */
+    public void testDefaultMEsDifferent() {
+        assertNotSame(defaultME, defaultME2,
+                "Default injection points \"@Inject ManagedExecutor exec;\" should get different instances");
+    }
+
+    /**
+     * Verify the container creates a ManagedExecutor instance per unqualified injection point w/ config
+     */
+    public void testUnnamedConfigDifferent() {
+        assertNotSame(defaultAnno1, defaultAnno2,
+                "Default injection points with matching @ManagedExecutorConfig should get different instances");
+    }
+
+    /**
+     * Verify injection points with matching @NamedInstance qualifiers share an ME instance
+     */
+    public void testNamedExecsSame() {
+        assertEquals(namedExecA1, namedExecA2,
```

Comment:
This isn't really testing that the injection points share an instance, just that the instances they return are .equals to each other.  This should be switched to assertSame

Reply:
both good points, do we think it is sufficient to just verify that the ManagedExecutor toString()'s are equal?
It looks like the proxies do return the toString of the underlying objects -- for example:
`com.ibm.ws.concurrent.mp.ManagedExecutorImpl@66ab0056 `

Reply:
Yes I think `toString()` will work so long as implementation is requires to override it (which is one of the reasons why I originally wanted to have that) and we already have a PR for that.
As I stated in https://github.com/eclipse/microprofile-concurrency/pull/92#discussion_r249139741, CDI forwards `toString()` invocations to the actual underlying instance so long as the implementation overrides it. What I forgot to add is that since that time I verified it manually as well :)